### ISSUE TITLE
Check if preview feature "Declarative user profile" can be supported by default

### ISSUE THREAD

**Issue by mposolda** [keycloak/keycloak#16741]
### Description

See the checklist on https://github.com/keycloak/keycloak/issues/16737

**Comment by pedroigor** [comment]
@stianst @mposolda To make sure we are aligned about how we are going to enable user profile:

* Existing realms are going to use legacy so that admins can choose whether they want to move to the declarative user profile
* New realms are going to be created with declarative user profile enabled

I'm not sure if we are going to remove the legacy user profile support anytime soon because FTL templates have changed a lot in order to support all the dynamics from the declarative user profile.

Once we enable it as the default, we should also gather a lot more feedback than what we had so far so that we can polish the capability and feel more comfortable about removing legacy.

In addition to that, we don't have a clear migration for users extending our themes and how they should adapt their customizations to the declarative user profile. That is probably one of the main gaps we have and I would like to fill it accordingly.

In terms of testing, we have good coverage and once we make it the default, the whole test suite will be running on top of it. Today, the declarative user profile is tested against some key functionalities like:

* Brokering
* Registration
* Account Management
* Admin Management
* Required Actions
* AIA

Basically, the main places where the user profile is involved.

In terms of documentation, we have https://www.keycloak.org/docs/latest/server_admin/index.html#user-profile. We should also review it accordingly to make sure all aspects are covered.
